src/network.c

```diff
@@ -645,7 +616,8 @@ int write_ssl(SSL* ssl, const char* buf, int amount) {
  * @param conn the Connection
  * @param buf buffer with data to write
  * @param amount the size of the data
- * @return The amount of bytes written to the Connection
+ * @return The amount of bytes written to the Connection.
+ *         -1 when it gets an unrecoverable error, just like write().
  */
 int writen_any(Connection* conn, const void* buf, int amount) {
```

Comment:
This loop will be exited either by all the data being written OR by `write()` getting an unrecoverable error. Are we worried about `write()` getting recoverable errors in an infinite loop?